tests/test_data_generator.py

```diff
@@ -0,0 +1,121 @@
+from copy import deepcopy
+
+import numpy as np
+import pytest
+
+from ssms.dataset_generators.lan_mlp import data_generator
+from ssms.config import model_config, data_generator_config
+
+
+def test_data_generator(tmp_path):
+    # Check included models
+    assert list(model_config.keys())[:10] == [
+        "ddm",
+        "ddm_legacy",
+        "angle",
+        "weibull",
+        "levy",
+        "levy_angle",
+        "full_ddm",
+        "full_ddm_rv",
+        "ddm_st",
+        "ddm_truncnormt",
+    ]
+
+    # Take an example config for a given model
+    model_conf = model_config["ddm"]
+
+    assert type(model_conf["simulator"]).__name__ == "cython_function_or_method"
+
+    assert callable(model_conf["simulator"])
+    assert callable(model_conf["boundary"])
+
+    # Initialize the generator config (for MLP LANs)
+    gen_config = data_generator_config["lan"]
+    generator_config = deepcopy(gen_config)
+    # Specify generative model (one from the list of included models mentioned above)
+    generator_config["dgp_list"] = "angle"
```

Comment:
should we parameterize this?

Reply:
as a first pass ok to specialize to angle, but we probably want this runnable to a bunch of models (we can pick a subset that covers some notion of "kinds" later on)